File: springboot-backend/src/main/java/org/example/springbootbackend/service/StoreItemService.java
package org.example.springbootbackend.service;

import org.example.springbootbackend.dto.StoreItemDto;
import org.example.springbootbackend.entity.StoreItem;

import java.util.List;

public interface StoreItemService {

    public StoreItemDto postStoreItem(StoreItemDto storeItemDto);
    public List<StoreItemDto> getAllStoreItem();
    public StoreItemDto getStoreItemById(Long id);
    public StoreItemDto updateProduct(Long id, StoreItemDto storeItemDto);
    public Boolean deleteStoreItem(Long id);

}

File: angular-frontend/src/app/Service/store-item.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8086/api/";
@Injectable({
  providedIn: 'root'
})
export class StoreItemService {

  constructor(private http : HttpClient) { }

  //create Product
  postStoreItem(storeItem:any):Observable<any>{
    return this.http.post(BASIC_URL +"store/storeItem",storeItem);
  }

  //get ALl product
  getAllStoreItem():Observable<any>{
    return this.http.get(BASIC_URL + "store/storeItem")
  }

  //Delete Product
  deleteStoreItem(id:number):Observable<any>{
    return this.http.delete(BASIC_URL + "store/storeItem/" + id)
  }

  //get Product By id 
  getStoreItemById(id:number):Observable<any>{
    return this.http.get(BASIC_URL + "store/storeItem/" + id)
  }

  //Update Product
  updateStoreItem(id:number, storeItem:any):Observable<any>{
    return this.http.put(BASIC_URL + "store/storeItem/" + id , storeItem);
  }
}
